src/lib/utils.js

```diff
@@ -235,6 +245,7 @@ function getRefinements(results, state, clearsQuery) {
     clearsQuery && state.query && state.query.trim()
       ? [
           {
+            attributeName: 'query',
```

Comment:
We can add a test case for this change since it's an issue on the current implementation.

Reply:
Added tests.